File: Corte 2/Taller 1/Burbuja.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define N 1000

void bubbleSort(int arr[], int n, int *comparisons, int *swaps) {
    int i, j, temp;
    for (i = 0; i < n - 1; i++) {
        for (j = 0; j < n - i - 1; j++) {
            (*comparisons)++;
            if (arr[j] > arr[j + 1]) {
                // Swap
                temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
                (*swaps)++;
            }
        }
    }
}

int main() {
    int arr[N];
    int comparisons = 0, swaps = 0;

    // Generar números aleatorios de 0 a 9
    srand(time(0));
    for (int i = 0; i < N; i++) {
        arr[i] = rand() % 10; // Números entre 0 y 9
    }

    // Mostrar números generados
    printf("Numeros generados:\n");
    for (int i = 0; i < N; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    // Ordenar usando el algoritmo de burbuja
    bubbleSort(arr, N, &comparisons, &swaps);

    // Mostrar números ordenados
    printf("Numeros ordenados:\n");
    for (int i = 0; i < N; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    // Mostrar resultados
    printf("Comparaciones: %d\n", comparisons);
    printf("Intercambios: %d\n", swaps);
    printf("Estable: Si\n");
    printf("Insercion: No\n");

    return 0;
}

File: Corte 2/Taller 1/Algoritmo de Ordenamiento Python/burbuja.py
import random
import time

# Función de ordenamiento burbuja
def bubble_sort(arr, comparisons, swaps):
    n = len(arr)
    for i in range(n - 1):
        for j in range(n - i - 1):
            comparisons[0] += 1
            if arr[j] < arr[j + 1]:
                # Intercambio
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swaps[0] += 1

def main():
    N = 1000
    arr = [0] * N
    comparisons = [0]  # Usamos una lista para modificar dentro de la función
    swaps = [0]

    # Medir el tiempo de ejecución
    start_time = time.perf_counter_ns()

    # Generar números aleatorios de 0 a 9
    for i in range(N):
        arr[i] = random.randint(300000,302500)  # Números entre 300000 y 302500

    # Mostrar números generados
    print("Números generados:")
    print(" ".join(map(str, arr)))

    # Ordenar usando el algoritmo de burbuja
    bubble_sort(arr, comparisons, swaps)

    end_time = time.perf_counter_ns()

    # Calcular el tiempo transcurrido en nanosegundos
    time_taken = end_time - start_time

    # Mostrar números ordenados
    print("\nNúmeros ordenados:")
    print(" ".join(map(str, arr)))

    # Mostrar resultados
    print(f"\nComparaciones: {comparisons[0]}")
    print(f"Intercambios: {swaps[0]}")
    print("Estable: Sí")
    print("Inserción: No")

    # Mostrar el tiempo de ejecución en nanosegundos
    print(f"Tiempo de ejecución en nanosegundos: {time_taken}")

if __name__ == "__main__":
    main()
